### PR TITLE
HYDRA-1071 : Add a flow viewport example that uses a custom GLSLFX shader for Storm

### PR DESCRIPTION
**Pull request by lanierd-adsk**

### CODE REVIEW

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+else
+(
+	if (IS_LINUX)
+		configure_file("plugInfo_Linux.json" ${CMAKE_CURRENT_BINARY_DIR}/plugInfo.json)
+	endif()
+	if (IS_MACOSX)
+		configure_file("plugInfo_OSX.json" ${CMAKE_CURRENT_BINARY_DIR}/plugInfo.json)
+	endif()
+)
+endif()
```
Comment (debloip-adsk) [inline lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt]: Minor : Wouldn't it be more consistent if we treated all platforms the same way? i.e. I don't see why the `if (IS_WINDOWS)` case is handled separately from Linux and OSX

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/README [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This is a pure usd plugin that is a shaders discovery plugin.
+The shaders defined inshadersDef.usda will be loaded by usd/hydra and available for use in a material network in any Hydra plugin.
```
Comment (debloip-adsk) [inline lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/README]: Typo : missing space in "`inshadersDef.usda`"

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
```
Comment (debloip-adsk) [inline lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.cpp:23]: Why the Pixar license?

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+// You may obtain a copy of the Apache License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the Apache License with the above modification is
+// distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the Apache License for the specific
+// language governing permissions and limitations under the Apache License.
+//
```
Comment (debloip-adsk) [inline lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.h:23]: Why the Pixar license?

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndexUtils.h [hunk 5]
```diff
@@ -24,8 +24,10 @@
 #include <pxr/imaging/hd/materialNetworkSchema.h>
 #include <pxr/imaging/hd/materialNodeSchema.h>
 #include <pxr/imaging/hd/materialSchema.h>
+#include <pxr/imaging/hd/perfLog.h>//For HD_TRACE_FUNCTION
```
Comment (debloip-adsk) [inline lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndexUtils.h:27]: This seems unused

**Review by debloip-adsk — CHANGES_REQUESTED** [review]
Looks good, thanks! Only minor touchups and questions

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt (hunk 1 above)
Comment (lanierd-adsk) [reply lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt]: I cannot do this or haven't found a way to do so. The name of the dynamic library is different under windows/Linx and OSX and havcing a single file like I did originally works only on Windows.

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.cpp (hunk 3 above)
Comment (lanierd-adsk) [reply lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.cpp:23]: It comes from a Pixar's file from their examples provided with USD, This license was left also in the file from Fusion, so I left it as well.

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.h (hunk 4 above)
Comment (lanierd-adsk) [reply lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/fvpShadersDiscoveryPlugin.h:23]: Same as above, it's a file originally made by Pixar.

File: lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndexUtils.h (hunk 5 above)
Comment (lanierd-adsk) [reply lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndexUtils.h:27]: It's actually used for HD_TRACE_FUNCTION, removing it doesn't compile on different platforms.

File: lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt (hunk 1 above)
Comment (debloip-adsk) [reply lib/flowViewport/usdPlugins/shadersDiscoveryPlugin/CMakeLists.txt]: Ah sorry I meant just the CMake if cases, not the plugInfo files themselves. Just that I don't see why the Windows and OSX/Linux cases are nested differently. But it is a nitpick, though I see you've changed it

**Review by debloip-adsk — APPROVED** [review]

**Pull request merged**